src/utils/search.ts

```diff
@@ -18,23 +19,44 @@ export const buildWhereClause = async (
   if (name) {
     const nameKeywords = name.split(" ");
     whereClause.productName = {
-      [Op.or]: [
-        { [Op.iLike]: `%${name}%` },
-        {
-          [Op.and]: nameKeywords.map((keyword) => ({
-            [Op.iLike]: `%${keyword}%`
-          }))
-        }
-      ]
+      [Op.and]: nameKeywords.map((keyword) => ({
+        [Op.iLike]: `%${keyword}%`
+      }))
     };
   }
 
   if (minPrice !== undefined && maxPrice !== undefined) {
+    const productsExist = await Product.findOne({
+      where: {
+        productPrice: {
+          [Op.between]: [minPrice, maxPrice]
+        }
+      }
+    });
+
+    if (!productsExist) {
+      throw new Error("No products found within the specified price range");
+    }
+
     whereClause.productPrice = {
       [Op.between]: [minPrice, maxPrice]
     };
   } else {
     if (minPrice !== undefined) {
+      const minPriceProducts = await Product.findAll({
+        where: {
```

Comment:
@PrinceRWIGIMBA it appears that you are not sending one request on database. The idea was to create a single object with query attributes and send that with one db request.

Reply:
i have updated my code